lib/location/method_wifi.c

```diff
@@ -0,0 +1,307 @@
+
+static uint32_t current_scan_result_count;
+static uint32_t latest_scan_result_count;
+
+struct method_wifi_scan_result {
+	char mac_addr_str[WIFI_MAC_ADDR_STR_LEN + 1];
+	char ssid_str[WIFI_SSID_MAX_LEN + 1];
+	uint8_t channel;
+	int8_t rssi;
+};
+
+static struct method_wifi_scan_result
+	latest_scan_results[CONFIG_LOCATION_METHOD_WIFI_SCANNING_RESULTS_MAX_CNT];
+static K_SEM_DEFINE(wifi_scanning_ready, 0, 1);
+
+/******************************************************************************/
+
+static int method_wifi_scanning_start(void)
+{
+	int ret;
+
+	LOG_DBG("Triggering start of WiFi scanning");
+
+	latest_scan_result_count = 0;
+	current_scan_result_count = 0;
+
+	if (wifi_iface != NULL) {
+		ret = net_mgmt(NET_REQUEST_WIFI_SCAN, wifi_iface, NULL, 0);
+		if (ret) {
+			LOG_ERR("Failed to initiate WiFi scanning: %d", ret);
+			ret = -1;
+		}
+	} else {
+		LOG_ERR("Failed to initiate WiFi scanning: no wifi device");
+		ret = -1;
+	}
+	return ret;
+}
+
+/******************************************************************************/
+
+static void method_wifi_handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_scan_result *entry = (const struct wifi_scan_result *)cb->info;
+	struct method_wifi_scan_result *current;
+
+	current_scan_result_count++;
+	current = &latest_scan_results[current_scan_result_count - 1];
+
+	if (current_scan_result_count <= CONFIG_LOCATION_METHOD_WIFI_SCANNING_RESULTS_MAX_CNT) {
+		sprintf(current->mac_addr_str,
+			"%02x:%02x:%02x:%02x:%02x:%02x",
+			entry->mac[0], entry->mac[1], entry->mac[2],
+			entry->mac[3], entry->mac[4], entry->mac[5]);
+		snprintf(current->ssid_str, entry->ssid_length + 1, "%s", entry->ssid);
+
+		current->channel = entry->channel;
+		current->rssi = entry->rssi;
+
+		LOG_DBG("scan result #%d stored: ssid %s, mac address: %s, channel %d,",
+			current_scan_result_count,
+			log_strdup(current->ssid_str),
+			log_strdup(current->mac_addr_str),
+			current->channel);
+	} else {
+		LOG_WRN("Scanning result (mac %02x:%02x:%02x:%02x:%02x:%02x) "
+			"did not fit to result buffer - dropping it",
+				entry->mac[0], entry->mac[1], entry->mac[2],
+				entry->mac[3], entry->mac[4], entry->mac[5]);
+	}
+}
+
+static void method_wifi_handle_wifi_scan_done(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+
+	if (status->status) {
+		LOG_WRN("WiFi scan request failed (%d).", status->status);
+	} else {
+		LOG_INF("Scan request done.");
+	}
+	k_sem_give(&wifi_scanning_ready);
+
+	latest_scan_result_count =
+		(current_scan_result_count > CONFIG_LOCATION_METHOD_WIFI_SCANNING_RESULTS_MAX_CNT) ?
+			CONFIG_LOCATION_METHOD_WIFI_SCANNING_RESULTS_MAX_CNT :
+			current_scan_result_count;
+	current_scan_result_count = 0;
+}
+
+static struct net_mgmt_event_callback method_wifi_net_mgmt_cb;
+
+void method_wifi_net_mgmt_event_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+					struct net_if *iface)
```

Comment:
mgmt_event could be const argument. Also iface don't seem to be used anywhere? How does this even compile without a warning?

Reply:
This is a handler given as function pointer so we cannot decide the parameters but they come from net_mgmt.h:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/net/net_mgmt.h#L113

mgmt_event is uint32_t which is basic non-pointer type. These are traditionally not marked as const.

Reply:
Add `ARG_UNUSED(iface);`?

Reply:
What is open from @hiltunent comment? 
ARG_UNUSED(iface) is there and @trantanen already said that: "mgmt_event is uint32_t which is basic non-pointer type. These are traditionally not marked as const." And that is how it is elsewhere.

Please, reopen if needed.